src/test/java/ua/tc/marketplace/service/impl/CategoryServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+package ua.tc.marketplace.service.impl;
+
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import ua.tc.marketplace.exception.category.CategoryNotFoundException;
+import ua.tc.marketplace.model.dto.category.CategoryDTO;
+import ua.tc.marketplace.model.dto.category.CreateCategoryDTO;
+import ua.tc.marketplace.model.dto.category.UpdateCategoryDTO;
+import ua.tc.marketplace.model.entity.Category;
+import ua.tc.marketplace.model.entity.ClassificationAttribute;
+import ua.tc.marketplace.repository.CategoryRepository;
+import ua.tc.marketplace.repository.ClassificationAttributeRepository;
+import ua.tc.marketplace.service.impl.CategoryServiceImpl;
+import ua.tc.marketplace.util.mapper.CategoryMapper;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+class CategoryServiceImplTest {
+    @Mock
+    private ClassificationAttributeRepository attributeRepository;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private CategoryMapper categoryMapper;
+
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+
+    @Test
+    void findCategoryById_shouldFindCategory_whenExists() {
+        Long categoryId = 1L;
+        Category category = new Category(); // create a sample Category entity
+        CategoryDTO categoryDto = new CategoryDTO(); // create a sample CategoryDTO
+
+        when(categoryRepository.findById(categoryId)).thenReturn(Optional.of(category));
+        when(categoryMapper.toCategoryDto(category)).thenReturn(categoryDto);
+
+        CategoryDTO result = categoryService.findById(categoryId);
+
+        assertEquals(categoryDto, result);
+        verify(categoryRepository, times(1)).findById(categoryId);
+        verify(categoryMapper, times(1)).toCategoryDto(category);
+    }
+
+    @Test
+    void findCategoryById_shouldThrow_whenNotExists() {
+        Long categoryId = 1L;
+
+        when(categoryRepository.findById(categoryId)).thenReturn(Optional.empty());
+
+        assertThrows(CategoryNotFoundException.class, () -> categoryService.findById(categoryId));
+
+        verify(categoryRepository, times(1)).findById(categoryId);
+        verify(categoryMapper, never()).toCategoryDto(any(Category.class));
+    }
+
+    @Test
+    void createCategory_shouldCreate_whenValidInput() {
```

Comment:
this test fails `java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because "attributeIds" is null`